packages/icons/scripts/build.js

```diff
@@ -0,0 +1,100 @@
+const path = require("path");
+const {
+  existsSync,
+  mkdirSync,
+  readdirSync,
+  readFile,
+  readFileSync,
+  writeFileSync
+} = require("fs");
+const SVGO = require("svgo");
+const svgstore = require("svgstore");
+const iconSpriteConfig = require("../iconSpriteConfig.js");
+const svgo = new SVGO({
+  plugins: [
+    {
+      removeDoctype: true,
+    },
+    {
+      cleanupIDs: false,
+    }
+  ]
+});
+const buildDirPath = path.join(__dirname, "../", "dist");
+
+const getFilePaths = dir =>
+  readdirSync(dir).map(file => `${dir}/${file}`);
+
+const writeSprite = (srcDir, spritePath, idPrefix) => {
+  console.info(`\tgenerating sprite at:\n\t${spritePath}\n`);
+
+  writeFileSync(
+    spritePath,
+    getFilePaths(srcDir).reduce((sprites, file) =>
+      sprites.add(`${idPrefix}-${path.basename(file, ".svg")}`, readFileSync(file, "utf8")),
+      svgstore({renameDefs: true})
+    )
+  );
+};
+
+const optimizeWithSVGO = spritePath => {
+  console.info("\toptimizing sprite\n");
+
+  readFile(spritePath, "utf8", (err, data) => {
+    if (err) {
+      throw err;
+    }
+
+    svgo.optimize(data, { path: spritePath }).then(result => {    
+      writeFileSync(spritePath, result.data);
+    });
+  });
+};
+
+const writeEnum = (srcDir, iconSetName, idPrefix) => {
```

Comment:
I wonder if @DanielMSchmidt would have any idea on how we could improve code generator?

Reply:
I think we could go the other route and get rid of the quotes problem if we did something like this:


```js
const t = require("babel-types");
const generate = require("@babel/generator");

const ast = t.tsEnumDeclaration(
  t.identifier(`${iconSetName.replace(/^\w/, c => c.toUpperCase())}Icons`),
  Object.keys(svgNamesObj).map(curr =>
    t.tsEnumMember(`${idPrefix}-${svgNamesObj[curr]}"`)
  )
);
const string = generate(ast);
```